lib/actions/user.js

```diff
@@ -0,0 +1,115 @@
+import { createAction } from 'redux-actions'
+
+import { addUser, fetchUser, updateUser } from '../util/middleware'
+
+const setCurrentUser = createAction('SET_CURRENT_USER')
+
+function getStateForNewUser (auth0User) {
+  return {
+    auth0UserId: auth0User.sub,
+    email: auth0User.email,
+    hasConsentedToTerms: false, // User must agree to terms.
+    isEmailVerified: auth0User.email_verified,
+    notificationChannel: 'email',
+    phoneNumber: '',
+    recentLocations: [],
+    savedLocations: [],
+    storeTripHistory: false // User must opt in.
+  }
+}
+
+export function isNewUser (loggedInUser) {
+  return !loggedInUser.hasConsentedToTerms
+}
+
+/**
+ * Fetches user preferences or set initial values under state.user if no user has been loaded.
+ */
+export function ensureLoggedInUserIsFetched (auth) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { loggedInUser } = user
+
+    if (auth && !loggedInUser) {
```

Comment:
This method's body needs some comments and I think it could use a bit of refactoring once you change `fetchUser` to be an action. `fetchUser` can handle setting the current user/accessToken.

EDIT: Actually, I think the right approach is to move the logic that checks auth and loggedInUser into `UserLoaderScreen#componentDidUpdate`. You can remove async/await from that and just perform these checks in the body of that lifecycle method. It can then fetch the user if the conditions are met.